tests/h/views/api/helpers/links_test.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import pytest
+
+from h.views.api.helpers import links
+
+
+class TestRegisterLink(object):
+    def test_it_creates_attrs_on_registry_if_not_present(
+        self, service_link, versions, pyramid_config
+    ):
+        links.register_link(service_link(), versions, pyramid_config.registry)
+
+        assert hasattr(pyramid_config.registry, "api_links")
+        assert "v1" in pyramid_config.registry.api_links
+        assert "v2" in pyramid_config.registry.api_links
+
+    def test_it_registers_link_for_every_version(
+        self, service_link, versions, pyramid_config
+    ):
+        link = service_link()
+        links.register_link(link, versions, pyramid_config.registry)
+        assert link in pyramid_config.registry.api_links["v1"]
+        assert link in pyramid_config.registry.api_links["v2"]
+
+    def test_it_does_not_register_link_for_unsupported_versions(
+        self, service_link, versions, pyramid_config
+    ):
+        first_service = service_link()
+        second_service = service_link("doodad")
+
+        links.register_link(first_service, versions, pyramid_config.registry)
+        links.register_link(second_service, ["v1"], pyramid_config.registry)
+
+        assert first_service in pyramid_config.registry.api_links["v2"]
+        assert second_service not in pyramid_config.registry.api_links["v2"]
+
+    def test_it_registers_primary_method_on_link(
+        self, service_link, versions, pyramid_config
+    ):
+        link = service_link()
+
+        links.register_link(link, versions, pyramid_config.registry)
+
+        assert pyramid_config.registry.api_links["v1"][0]["method"] == "POST"
+
+    def test_it_registers_first_primary_method_if_multiple(
+        self, service_link, versions, pyramid_config
+    ):
+        multi_link = service_link()
+        multi_link["method"] = ("GET", "POST")
+
+        links.register_link(multi_link, versions, pyramid_config.registry)
+
+        assert pyramid_config.registry.api_links["v1"][0]["method"] == "GET"
+
+
+@pytest.fixture
+def service_link():
+    def make_service_link(name=None):
+        return {
+            "name": name or "api.example_service",
```

Comment:
I think it'd be more idiomatic if the default value for this argument were `def make_service_link(name="api.example_service")` in the method signature rather than doing `name or "api.example_service"` in the method body

Reply:
Oh, sure! That's fine.